### ISSUE TITLE
EOFError: Compressed file ended before the end-of-stream marker was reached

### ISSUE THREAD

**Issue by Chuabacca** [apple/turicreate#2975]
My internet connection was interrupted while downloading the one_shot_object_detector backgrounds. Now when I try to train a model, I get this error. I think the issues is a corrupt gzip file, but I don't know where it is located.
```
---------------------------------------------------------------------------
EOFError                                  Traceback (most recent call last)
<ipython-input-26-307cbd9c95bb> in <module>
----> 1 model = tc.one_shot_object_detector.create(training_images, 'label')

~/venv/lib/python3.7/site-packages/turicreate/toolkits/one_shot_object_detector/one_shot_object_detector.py in create(data, target, backgrounds, batch_size, max_iterations, verbose)
     62         raise TypeError("'data' must be of type SFrame or Image.")
     63     augmented_data = _preview_synthetic_training_data(
---> 64         data, target, backgrounds)
     65     model = _tc.object_detector.create(augmented_data,
     66                                        batch_size=batch_size,

~/venv/lib/python3.7/site-packages/turicreate/toolkits/one_shot_object_detector/util/_augmentation.py in preview_synthetic_training_data(data, target, backgrounds, verbose, **kwargs)
     51         backgrounds_tar_path = backgrounds_downloader.get_backgrounds_path()
     52         backgrounds_tar = _tarfile.open(backgrounds_tar_path)
---> 53         backgrounds_tar.extractall()
     54         backgrounds = _tc.SArray("one_shot_backgrounds.sarray")
     55         # We resize the background dimensions by half along each axis to reduce

~/venv/lib/python3.7/tarfile.py in extractall(self, path, members, numeric_owner)
   2000             # Do not set_attrs directories, as we will do that further down
   2001             self.extract(tarinfo, path, set_attrs=not tarinfo.isdir(),
-> 2002                          numeric_owner=numeric_owner)
   2003 
   2004         # Reverse sort directories.

~/venv/lib/python3.7/tarfile.py in extract(self, member, path, set_attrs, numeric_owner)
   2042             self._extract_member(tarinfo, os.path.join(path, tarinfo.name),
   2043                                  set_attrs=set_attrs,
-> 2044                                  numeric_owner=numeric_owner)
   2045         except OSError as e:
   2046             if self.errorlevel > 0:

~/venv/lib/python3.7/tarfile.py in _extract_member(self, tarinfo, targetpath, set_attrs, numeric_owner)
   2112 
   2113         if tarinfo.isreg():
-> 2114             self.makefile(tarinfo, targetpath)
   2115         elif tarinfo.isdir():
   2116             self.makedir(tarinfo, targetpath)

~/venv/lib/python3.7/tarfile.py in makefile(self, tarinfo, targetpath)
   2161                 target.truncate()
   2162             else:
-> 2163                 copyfileobj(source, target, tarinfo.size, ReadError, bufsize)
   2164 
   2165     def makeunknown(self, tarinfo, targetpath):

~/venv/lib/python3.7/tarfile.py in copyfileobj(src, dst, length, exception, bufsize)
    245     blocks, remainder = divmod(length, bufsize)
    246     for b in range(blocks):
--> 247         buf = src.read(bufsize)
    248         if len(buf) < bufsize:
    249             raise exception("unexpected end of data")

/usr/local/Cellar/python/3.7.6_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/gzip.py in read(self, size)
    274             import errno
    275             raise OSError(errno.EBADF, "read() on write-only GzipFile object")
--> 276         return self._buffer.read(size)
    277 
    278     def read1(self, size=-1):

/usr/local/Cellar/python/3.7.6_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/_compression.py in readinto(self, b)
     66     def readinto(self, b):
     67         with memoryview(b) as view, view.cast("B") as byte_view:
---> 68             data = self.read(len(byte_view))
     69             byte_view[:len(data)] = data
     70         return len(data)

/usr/local/Cellar/python/3.7.6_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/gzip.py in read(self, size)
    480                 break
    481             if buf == b"":
--> 482                 raise EOFError("Compressed file ended before the "
    483                                "end-of-stream marker was reached")
    484 

EOFError: Compressed file ended before the end-of-stream marker was reached
```

**Comment by Chuabacca** [comment]
Restarting my computer solved the issue!

**Comment by TobyRoseman** [comment]
@Chuabacca - I'm glad to hear that restarting your computer solved the problem. That makes sense and thanks for letting us know.

You could also have run the following script:
```python
from turicreate.toolkits._pre_trained_models import _get_cache_dir
print(_get_cache_dir('data'))
```
Then deleted the directory that gets printed out. 

FYI - the underlying issue was fixed by #2901 and that fix will be included in our next release.

**Comment by Chuabacca** [comment]
Great, thanks for the info! And glad to hear that the underlying issue was addressed. I’m enjoying learning and using turi create.
